added general mqtt concepts (#7)
mqtt-varroa/varroa-documentation: LaTeX/Parts/Chapter2.tex

A Varroa distributed system contains at least one Agent.

\paragraph{Varroa Instance}

> A Varroa Instance is a running Varroa process in a single JVM, can be either Commander or Agent.

Reviewer: ```suggestion
A Varroa instance is a running Varroa process in a single JVM, it can either be a Commander or an Agent.
```

%A scenario is a XML-Document which defines a sequence of stages.

\section{General MQTT Concepts}
\paragraph{Broker}

> The Broker serves as an intermediary between publishers and subscribers, takes over the routing of the exchanged messages and is the central

Reviewer: ```suggestion
The Broker serves as an intermediary between publishers and subscribers. It takes over the routing of the exchanged MQTT messages and is the central
```

> control authority of an MQTT-Network.

Reviewer: ```suggestion
control authority of a MQTT-Network.
```

%TODO cite Georg
\paragraph{Client}

> A client that implements the MQTT protocol. 

Reviewer: add description  of what a client can be

> Can be publisher or subscriber.

Reviewer: remove line

\paragraph{Topic}
Topics are strings separated by slashes that do not contain wild-cards.

> Messages in a Topic are delivered to subscribers with matching Topic Filters.

Reviewer: ```suggestion
Messages published with a Topic are delivered to subscribers that have registered matching Topic Filters.
```

\paragraph{Topic Filter}
A Topic Filter is a chain of strings delimited by slashes that 
can cover one or more topics. 
It can also contain wild-cards: a plus sign covers a single hierarchy level, a double cross selects all possible following levels.
\paragraph{Publisher}
Publishers are clients that produce data.

> They send messages to a specified topic on the broker.

Reviewer: ```suggestion
They send messages with a specific topic to the broker.
```

\paragraph{Subscriber}
Subscribers are clients that subscribe to a subset or to all messages sent via the
MQTT network.
They log on to the broker and register with one or more topic filters that specify the topics whose messages they want to receive.
%TODO cite Georg
\paragraph{Connect}
Connect describes the process of a Client terminating its connection to a Broker.
\paragraph{Disconnect}

> Disconnect describes the process of a Client establishing a connection to a Broker.

Reviewer: connect != disconnect
